libbeat/autodiscover/template/config.go

```diff
@@ -123,11 +126,20 @@ func ApplyConfigTemplate(event bus.Event, configs []*common.Config, options ...u
 	if err != nil {
 		logp.Err("Error building config: %v", err)
 	}
+
 	opts := []ucfg.Option{
 		ucfg.PathSep("."),
 		ucfg.Env(vars),
 		ucfg.ResolveEnv,
 		ucfg.VarExp,
+
+		// Catch-all resolve function to log fields not resolved in any other way.
+		ucfg.Resolve(func(name string) (string, parse.Config, error) {
```

Comment:
Has this option to be added at the very end of the options list? If so, maybe it can be added after line 144 where extra options (related to keystore) are added?

Reply:
I thought the same and initially implemented it this way, but then I saw that this resolver was always executed, so it seems that they are executed in inverse order to the one they are added.

This seems to confirm that: https://github.com/elastic/go-ucfg/blob/bbb885db44c36f545b8a57b6a74ff0d7b09a37c6/variables.go#L150

Though this makes me think that I should then be adding it before `ResolveEnv` too.

Reply:
👍  it sounds sane to me!

Reply:
Yep, it should be added before `ResolveEnv`, I have reordered options. Thanks for pointing to this!

Reply:
Oh, and by adding it the first, its error is the one returned by `Unpack`, much better, I will remove the other debug log.